### ISSUE TITLE
Strip Invalid Hostname Characters from App Name When Tagging Images

### ISSUE THREAD

**Issue by rringler** [convox/rack#967]
Feel free to close this if it's dumb for other reasons, but I just used Convox to dockerize a sample rails app, which had trouble connecting to its postgres process.  A little troubleshooting showed it was because I had created the app in a `sample_app` directory, which Convox used for docker image names, resulting in an invalid URI (illegal underscore) in my `DATABASE_URL` envvar.

Maybe Convox should strip out invalid hostname characters when tagging its docker images?

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;GgVBVDQO&#x2F;481-invalid-hostname-characters"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Invalid Hostname Characters</a>

**Comment by MiguelMoll** [comment]
Hey @rringler. Thanks for bringing up this issue.

To be on the same page, you created a Rack app named `sample_app` and that resulted in a `DATABASE_URL` similar to `postgres://postgres:password@172.17.0.2:5432/sameple_app`?

**Comment by rringler** [comment]
Hi @MiguelMoll,

I created a new rails app named `sample_app`, and copied the `Dockerfile` and `docker-compose.yml` from the tutorials:

`Dockerfile`:
```
FROM convox/rails

COPY Gemfile      /app/Gemfile
COPY Gemfile.lock /app/Gemfile.lock

RUN bundle install

COPY Rakefile   /app/Rakefile
COPY config     /app/config
COPY public     /app/public

COPY . /app
```

`docker-compose.yml`:
```yaml
web:
  build: .
  labels:
    - convox.port.443.protocol=tls
    - convox.port.443.proxy=true
  links:
    - database
  ports:
    - 80:4000
    - 443:4001
database:
  image: convox/postgres
  ports:
    - 5432
```

This allows me to run the app via `$ convox start`, and docker shows the following running images:
```bash
$ docker ps
CONTAINER ID        IMAGE                 COMMAND                  CREATED             STATUS              PORTS                                           NAMES
be7b2bd5e9ce        convox/proxy          "proxy-link 443 4001 "   15 seconds ago      Up 14 seconds       0.0.0.0:443->443/tcp                            drunk_leavitt
324e574e715a        sample_app/web        "bin/web"                16 seconds ago      Up 15 seconds       0.0.0.0:80->4000/tcp, 0.0.0.0:39141->4001/tcp   sample_app-web
d6bdee5e48c0        sample_app/database   "/docker-entrypoint.s"   17 seconds ago      Up 16 seconds       0.0.0.0:5432->5432/tcp                          sample_app-database
```

This is up an running, but as soon I try to connect to the rails app, Rails raises an exception:
```
URI::InvalidURIError (the scheme postgres does not accept registry part: postgres:password@sample_app-database:5432 (or bad hostname?))
```

I believe is due to the **`sample_app-database`** hostname in the URI, which [this stack overflow article](http://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it/2183140#2183140) mentions is illegal per RFC1123.

To test this, a simple `$ mv sample_app/ sampleapp/` resolves this issue and rails can talk to the database correctly.

**Comment by ddollar** [comment]
Fixed by #1040
